### PR TITLE
Add alembic migration

### PR DESCRIPTION
**Pull request by kitsuta**
Adds a working alembic configuration, plus a test migration that will add the bands and panels plugin tables.

### CODE REVIEW

File: alembic/env.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+# Interpret the config file for Python logging.
+# This line sets up loggers basically.
+fileConfig(config.config_file_name)
+
+# add your model's MetaData object here
+# for 'autogenerate' support
+# from myapp import mymodel
+# target_metadata = mymodel.Base.metadata
+import sideboard
```
Comment (kitsuta) [inline alembic/env.py]: This is where we add the metadata from RAMS.

File: alembic/env.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+from uber.common import *
+target_metadata = Session.BaseClass.metadata
+
+# other values from the config, defined by the needs of env.py,
+# can be acquired:
+# my_important_option = config.get_main_option("my_important_option")
+# ... etc.
+# We replace the INI option with RAMS' URL.
+config.set_main_option('sqlalchemy.url', c.SQLALCHEMY_URL)
```
Comment (kitsuta) [inline alembic/env.py]: This is some fancy magic where we use RAMS' config option to replace the option in alembic.ini, saving us some trouble.

File: alembic/versions/8c6cddf6335d_add_panels_and_bands_plugins.py [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+# revision identifiers, used by Alembic.
+revision = '8c6cddf6335d'
+down_revision = None
+branch_labels = None
+depends_on = None
+
+from alembic import op
+import sqlalchemy as sa
+import sideboard
```
Comment (kitsuta) [inline alembic/versions/8c6cddf6335d_add_panels_and_bands_plugins.py]: This three import lines needed to be added manually, but I would be surprised if we couldn't do that automatically.

File: alembic/versions/8c6cddf6335d_add_panels_and_bands_plugins.py [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+import uber
+from uber import c
+
+def upgrade():
+    op.create_table('event',
+    sa.Column('id', sideboard.lib.sa.UUID(), nullable=False),
+    sa.Column('location', uber.models.Choice(c.EVENT_LOCATION_OPTS), nullable=False),
```
Comment (kitsuta) [inline alembic/versions/8c6cddf6335d_add_panels_and_bands_plugins.py]: Everywhere where you see a `c` option, that had to be added manually - alembic originally tried to add these columns with no Choice arguments! I don't know if that's easy to get around in a permanent sense.

**Comment by kitsuta** [comment]
Steps to make this work:
```
pip install alembic
cd ~/uber/sideboard/plugins/magstock
alembic upgrade head
```

To generate a new revision:
```
alembic revision --autogenerate -m "Message here"
```
New revisions are found in magstock/alembic/versions

Unfortunately, I cannot find a way to 'fast-forward' through revisions, so in order to upgrade, you'll need to start with no bands/plugins tables - otherwise it'll fail because some of the tables already exist.

I am sure that this can be improved further - alembic has branches, even! But I wanted to start with a basic proof of concept.

**Comment by binary1230** [comment]
OMHBITY +1 if you want to go for it.

I guess for now the plan is to have 3 different alembic setups in each of our 3 plugins like stock, classic, prime.

**Comment by kitsuta** [comment]
Yup, that's the idea.

Here's the real-world implementation - PR https://github.com/magfest/ubersystem/pull/1839 requires a new column, PLUS we need to migrate data in that column (filling in email_id). I've replaced the bands migration with this one. Merging now.

**Pull request merged**